test/setup

```diff
@@ -0,0 +1,74 @@
+set -e
+
+exec 1001> /dev/stdout
+BASH_XTRACEFD=1001
+echo "$0" "$@"
+set -x
+
+test -n "$MPV_MPRIS_TEST_NAME" ||
+exec "./env" "$0" "$@"
+
+test="$(basename "$0")"
+
+if [ "$test" = setup ] ; then
+	echo "This setup script should be sourced, not executed" >&2
+	exit 1
+fi
+
+ipc="$MPV_MPRIS_TEST_IPC/$test.mpv.ipc"
+log_prefix="$MPV_MPRIS_TEST_LOG/$test"
+log="$log_prefix.mpv.log"
+file="$MPV_MPRIS_TEST_PLAY"
+tty="$(tty)"
```

Comment:
I tried to run these tests as package tests on Gentoo, and failed here due to:
```
+++ tty                                                                                                               
++ tty='not a tty'                                                                                                                                                                                                                          
make[1]: *** [Makefile:6: test] Error 1                                                                                                                                                                                                     
```
For better portability of the test suite, it would of course be nice to be able to run without a `tty`. Maybe wrap everything in `script`?

Reply:
Alternatively, maybe use:
```
tee /dev/stderr
```
below, which drops the need for a `tty`.

Reply:
I'm trying to avoid stderr, since [Debian's CI system by default wants stderr to be empty](https://salsa.debian.org/ci-team/autopkgtest/-/raw/master/doc/README.package-tests.rst) (search for stderr) and I would like to abide by that in case later on there is an mpv message on stderr that indicates some breakage. For examples of that avoidance, see in the setup script were I redirect part of the mpv stderr to stdout and where I used `BASH_XTRACEFD` to redirect `set -x` output to stdout.

I tried `tee /dev/stdout` in the Debian CI and I didn't get any of the JSON strings printed somehow. Seems like that is because it resolves to the stdout of the tee process, which goes into the pipe, which is obviously wrong. So I probably need to either resolve `/dev/stdout` to something first, or just write to a file and then cat the file, which seems like the simplest solution. It seems like another option is process substitution, and since we are depending on bash for `BASH_XTRACEFD`, it seems fine to use process substitution for this.

Reply:
Process substitution ends up making the indentation etc a bit annoying, so I decided to go with just writing files and using input/output redirection. As a bonus the bash trace output is much more readable this way.

Reply:
> I'm trying to avoid stderr, since [Debian's CI system by default wants stderr to be empty](https://salsa.debian.org/ci-team/autopkgtest/-/raw/master/doc/README.package-tests.rst) (search for stderr) and I would like to abide by that in case later on there is an mpv message on stderr that indicates some breakage. 

Indeed, that's a good point (and I also learnt something new about Debian's `autopkgtest`s this way ;-) ). 

The solution you chose sounds great, process substitution is really a bit of a mess to use, Another idea could be named file descriptors, but there's not much of a difference.

Reply:
Thanks for the named file descriptors tip! That lets me get bash to allocate the `set -x` output file descriptor instead of choosing one and hoping it isn't in use :grinning: